ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/ReplayableCommandQueue.java

```diff
@@ -34,4 +35,6 @@ QueuedCommandStatus enqueueCommand(
   List<QueuedCommand> getNewCommands();
 
   List<QueuedCommand> getRestoreCommands();
+
+  CompletableFuture<Void> getConsumerPositionFuture(long offset);
```

Comment:
can't help thinking that it might be better  / more encapsulated to replace this method with one like:

```
void ensureConsumedUpto(long offset);
```

Which would encapsulate all the futures and waiting and exceptions etc...